pkg/cdn/metadata/mem/mem.go

```diff
@@ -0,0 +1,36 @@
+package mem
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/spf13/afero"
+
+	"github.com/gomods/athens/pkg/cdn/metadata"
+	"github.com/gomods/athens/pkg/cdn/metadata/fs"
+	"github.com/gomods/athens/pkg/errors"
+)
+
+var (
+	l          sync.Mutex
+	memStorage metadata.Storage
```

Comment:
Would you be able to avoid these package-level variables? it looks like you can create a new memory fs (`NewMemMapFs`) on each call to `NewStorage`. What do you think?

Reply:
The problem with that is that if we call NewStorage(...) and save something there we won't find it in another storage created with NewStorage(...) because it will reference different afero.Fs. with an OS file system it's not an issue but here we get a new instance on in-memory FS.